### PR TITLE
Add autoscaling support

### PR DESCRIPTION
**Pull request by enxebre**
Runs the autoscaler watching the target namespace.
Apply Autoscaling annotations to scalable resources based on nodepool.Spec.AutoScaling min/max.

Needs https://github.com/openshift/hypershift/pull/23

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/assets/autoscaler/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    matchLabels:
+      app: cluster-autoscaler
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cluster-autoscaler
+    spec:
+      containers:
+        - image: k8s.gcr.io/autoscaling/cluster-autoscaler:v1.20.0
```
Comment (derekwaynecarr) [inline hypershift-operator/controllers/hostedcluster/assets/autoscaler/deployment.yaml:18]: we will need a follow-up to pull this image from our release payload.

i think this is the first component where its not clear if we should version this with hypershift release or ocp release.

thoughts?

**Comment by ironcladlou** [comment]
/lgtm

@enxebre if this is ready, please remove the hold

**Pull request merged**